src/main/java/command/AddCommand.java

```diff
@@ -84,7 +72,7 @@ private String convertToPascalCaseNoSpace(String input) throws CommandException
 
             return String.join("", splitInput);
         } catch (StringIndexOutOfBoundsException s) {
-            throw new CommandException(" why extra space?");
+            throw new CommandException("Why is there extra space?");
```

Comment:
Should we add `.trim()` to avoid this?

Reply:
I believe the issue was for when this kind of input "many      spaces", the splitInput var will have empty entries of ""  which is you invoke substring(1) causes it to access an index out of bounds. Trying a different solution.